### PR TITLE
uipbuf: added some documentation

### PR DESCRIPTION
**Pull request by joakimeriksson**
Added some documentation - some of which indicate feature that can be added (and is implemented in our code-base).

### CODE REVIEW

File: os/net/ipv6/uipbuf.h [hunk 1]
```diff
@@ -35,33 +35,98 @@
 
 #include "contiki.h"
 
-/* Get the next header given the buffer - start indicates that this is
-   start of the IPv6 header - needs to be set to 0 when in an ext hdr */
-
+/* \brief          Get the next IPv6 header.
```
Comment (g-oikonomou) [inline os/net/ipv6/uipbuf.h]: I think you need double-star here for doxygen to generate HTML for this

File: os/net/ipv6/uipbuf.h [hunk 2]
```diff
@@ -35,33 +35,98 @@
...
+ *
+ */
 enum {
-  UIPBUF_ATTR_LLSEC_LEVEL,
-  UIPBUF_ATTR_LLSEC_KEY_ID,
-  UIPBUF_ATTR_INTERFACE_ID,
-  UIPBUF_ATTR_PHYSICAL_NETWORK_ID,
-  UIPBUF_ATTR_MAX_MAC_TRANSMISSIONS,
-  UIPBUF_ATTR_FLAGS,
+  UIPBUF_ATTR_LLSEC_LEVEL,      /* Control link layer security level. */
```
Comment (g-oikonomou) [inline os/net/ipv6/uipbuf.h]: I think (didn't test) you need `/**<` here for doxygen to generate HTML

**Comment by g-oikonomou** [comment]
Thanks :)

File: os/net/ipv6/uipbuf.h (hunk 2 above)
Comment (joakimeriksson) [reply os/net/ipv6/uipbuf.h]: Maybe I should try adding that then!

**Pull request merged**